test/integration/fleetscope/fleetscope_test.go

```diff
@@ -100,7 +100,8 @@ func TestFleetscope(t *testing.T) {
 						// Service Mesh Management
 						{
 							assert.Equal("MANAGEMENT_AUTOMATIC", gkeFeatureOp.Get("fleetDefaultMemberConfig.mesh.management").String(), fmt.Sprintf("Hub Feature %s should have mesh menagement equal to MANAGEMENT_AUTOMATIC", feature))
-							// TODO: validate clusters service mesh state
+							gkeMesh := gcloud.Runf(t, "beta container fleet mesh describe --project %s", clusterProjectID)
+							assert.Equal("ACTIVE", gkeMesh.Get("resourceState.state").String(), fmt.Sprintf("Hub Feature %s should have resource state equal to ACTIVE", feature))
```

Comment:
Hey @amandakarina - I think we should also check/retry on `controlPlaneManagement` and `dataPlaneManagement` states, as if they are still in `PROVISIONING` (as in this test run) there is a good chance they won't be ready by the app deployment.


```
TestFleetscope/development 2024-11-11T17:27:21Z command.go:100: Running command gcloud with args [container hub features describe servicemesh --project eab-gke-development-oh22 --format json]
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185: {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   "createTime": "2024-11-11T17:21:39.460742525Z",
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   "fleetDefaultMemberConfig": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:     "mesh": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:       "management": "MANAGEMENT_AUTOMATIC"
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:     }
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   },
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   "labels": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:     "goog-terraform-provisioned": "true"
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   },
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   "membershipSpecs": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:     "projects/47262381159/locations/us-central1/memberships/cluster-us-central1-development": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:       "mesh": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:         "management": "MANAGEMENT_AUTOMATIC"
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:       },
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:       "origin": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:         "type": "USER"
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:       }
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:     }
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   },
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   "membershipStates": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:     "projects/47262381159/locations/us-central1/memberships/cluster-us-central1-development": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:       "servicemesh": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:         "controlPlaneManagement": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:           "details": [
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:             {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:               "code": "REVISION_PROVISIONING",
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:               "details": "Provisioning: asm-managed"
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:             }
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:           ],
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:           "state": "PROVISIONING"
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:         },
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:         "dataPlaneManagement": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:           "details": [
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:             {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:               "code": "MANAGED_CONTROL_PLANE_REQUIRED",
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:               "details": "Requires active managed control plane."
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:             }
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:           ],
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:           "state": "PROVISIONING"
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:         }
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:       },
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:       "state": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:         "description": "Revision being rolled out: asm-managed.",
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:         "updateTime": "2024-11-11T17:26:32.767444344Z"
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:       }
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:     }
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   },
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   "name": "projects/eab-gke-development-oh22/locations/global/features/servicemesh",
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   "resourceState": {
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:     "state": "ACTIVE"
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   },
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   "spec": {},
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185:   "updateTime": "2024-11-11T17:25:51.903610973Z"
TestFleetscope/development 2024-11-11T17:27:22Z command.go:185: }
```